lib/quantization/src/encoded_vectors.rs

```diff
@@ -52,6 +52,15 @@ pub trait EncodedVectors: Sized {
     fn encode_internal_vector(&self, id: u32) -> Option<Self::EncodedQuery>;
 }
 
+pub trait EncodedVectorsBytes: EncodedVectors {
+    fn score_point_vs_bytes(
+        &self,
+        query: &Self::EncodedQuery,
+        bytes: &[u8],
+        hw_counter: &HardwareCounterCell,
+    ) -> f32;
+}
```

Comment:
We have a bunch of methods in the `EncodedVectors` trait, but not all quantizations implement all of the methods; some just `panic!` for unsupported methods. I don't like this approach, and for this new method I took a different one.

Instead of adding a new, potentially panicking method, I've added a new trait. Not all quantizations implement this trait, but those that do don't panic.

Not completely happy with this approach either, but let's see how would it go.